controllers/metal3.io/preprovisioningimage_controller.go

```diff
@@ -154,10 +155,10 @@ func buildIgnitionConfig(secret *corev1.Secret) ([]byte, error) {
 	}
 
 	builder := ignition.New(nmstate,
```

Comment:
Hi Angus, While testing with @imain and @squidboylan , we found that ignition embedded into the ISO must be a gzipped CPIO archive. Like this:
https://github.com/openshift/assisted-image-service/blob/main/internal/handlers/images.go#L289-L313
I think we're just adding a json byte array at the moment. Thanks.

Reply:
We're passing the json data to [`NewRHCOSStreamReader()`](https://github.com/openshift/image-customization-controller/blob/main/pkg/imagehandler/imagefilesystem.go#L63), which generates the CPIO archive internally using the code you linked.

Reply:
Ah, I see. In [the prototype](https://github.com/zaneb/iso-stream/blob/master/injector/injector.go#L18-L22) `NewRHCOSStreamReader()` took the raw file contents and turned it into a gzipped CPIO archive, but the version that landed in assisted-image-service expects that to have been done already :(

Reply:
https://github.com/openshift/assisted-image-service/pull/35

Reply:
Thanks @zaneb, i'll rebase as soon as that merges..